torchmetrics/metric.py

```diff
@@ -258,27 +269,55 @@ def sync(
                 only when running in a distributed setting.
             distributed_available: Function to determine if we are running inside a distributed setting
 
-        Returns:
-            cache: A dictionary containing the local metric states. The cache will be empty if sync didn't happen.
         """
+        if self.is_synced and should_sync:
+            raise TorchMetricsUserError("The Metric has already been synced.")
+
         is_distributed = distributed_available() if callable(distributed_available) else None
+
         if not should_sync or not is_distributed:
-            return {}
+            return
+
         if dist_sync_fn is None:
             dist_sync_fn = gather_all_tensors
+
         # cache prior to syncing
-        cache = {attr: getattr(self, attr) for attr in self._defaults}
+        self._cache = {attr: getattr(self, attr) for attr in self._defaults}
+
         # sync
         self._sync_dist(dist_sync_fn, process_group=process_group)
-        return cache
+        self.is_synced = True
+
+    def unsync(self, should_unsync: bool = True) -> None:
+        """
+        Unsync function for manually controlling when metrics states should be reverted back to their local states.
+
+        Args:
+            should_unsync: Whether to perform unsync
+
+        """
+        if not should_unsync:
+            return
+
+        if self.is_synced:
+            if self._cache is None:
+                raise TorchMetricsUserError("The internal cache should exist to unsync the Metric.")
+
+            # if we synced, restore to cache so that we can continue to accumulate un-synced state
+            for attr, val in self._cache.items():
+                setattr(self, attr, val)
+            self.is_synced = False
+            self._cache = None
+        else:
+            raise TorchMetricsUserError("The Metric has already been un-synced.")
```

Comment:
nit: prefer early returns/raises to avoid another level of nesting with `if/else` 
e.g. 
```
if not self.is_synced:
    raise error
# proceed 
```

Reply:
Done !